core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -565,10 +583,11 @@ public void forceAvailabilityMode(String cacheName, AvailabilityMode availabilit
       public void handleViewChange(final ViewChangedEvent e) {
          // Need to recover existing caches asynchronously (in case we just became the coordinator).
          // Cannot use the async notification thread pool, by default it only has 1 thread.
-         asyncTransportExecutor.submit(new Runnable() {
+         viewHandlingCompletionService.submit(new Callable<Void>() {
```

Comment:
Unfortunately, the way the `SemaphoreCompletionService` works, it would also need a return value with a `Runnable`, so it wouldn't make any difference. I'm considering making it an `ExecutorService` instead though, since I haven't really used the tracking of futures.